h2/src/main/org/h2/command/Parser.java

```diff
@@ -5956,6 +5957,17 @@ private DefineCommand parseAlterTableAddConstraintIf(String tableName,
                 read("BTREE");
             }
             return command;
+        } else if (allowAffinityKey && readIf("AFFINITY")) {
```

Comment:
I like the idea of having compatibility with other DBs (MemSQL at least). I think we have to check both keywords here: `AFFINITY` and `SHARD`. Probably lets create a method `readIfAffinity()` which will check them both.

Also it would be nice to have tests like this (taken from MemSQL docs):
```sql
    CREATE TABLE clicks (
        click_id BIGINT AUTO_INCREMENT,
        user_id INT,
        page_id INT,
        ts TIMESTAMP,
        SHARD KEY (user_id),
        PRIMARY KEY (click_id, user_id)
    );
```